tests/lib/rules/no-loop-func.js

```diff
@@ -742,3 +742,63 @@ ruleTester.run("no-loop-func", rule, {
 		},
 	],
 });
+
+const ruleTesterTypeScript = new RuleTester({
+	languageOptions: {
+		parser: require("@typescript-eslint/parser"),
+	},
+});
+
+ruleTesterTypeScript.run("no-loop-func", rule, {
+	valid: [
+		`
+  for (let i = 0; i < 10; i++) {
+	function foo() {
+	  console.log('A');
+	}
+  }
+	  `,
+		`
+  let someArray: MyType[] = [];
+  for (let i = 0; i < 10; i += 1) {
+	someArray = someArray.filter((item: MyType) => !!item);
+  }
+	  `,
+		{
+			code: `
+  let someArray: MyType[] = [];
+  for (let i = 0; i < 10; i += 1) {
+	someArray = someArray.filter((item: MyType) => !!item);
+  }
+		`,
+			languageOptions: {
+				globals: {
+					MyType: "readonly",
+				},
+			},
+		},
+		{
+			code: `
+  let someArray: MyType[] = [];
+  for (let i = 0; i < 10; i += 1) {
+	someArray = someArray.filter((item: MyType) => !!item);
+  }
+		`,
+			languageOptions: {
+				globals: {
+					MyType: "writable",
+				},
+			},
+		},
+		`
+  type MyType = 1;
+  let someArray: MyType[] = [];
+  for (let i = 0; i < 10; i += 1) {
+	someArray = someArray.filter((item: MyType) => !!item);
+  }
+	  `,
+	],
+	invalid: [
+		// Invalid TS code
```

Comment:
@snitin315 did you mean to push a commit with tests here?

Reply:
Also, I just noticed that in error message, we repeat the [same variable name in few cases](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tbG9vcC1mdW5jOiBcImVycm9yXCIqL1xuXG5mb3IgKHZhciBpID0gMDsgaSA8IDEwOyBpKyspIHtcbiAgaXRlbXMucHVzaCh7XG4gICAgaWQ6IGksXG4gICAgbmFtZTogXCJJdGVtIFwiICsgaVxuICB9KTtcblxuICBjb25zdCBwcm9jZXNzID0gZnVuY3Rpb24gKGNhbGxiYWNrKXtcbiAgICBjYWxsYmFjayh7IGlkOiBpLCBuYW1lOiBcIkl0ZW0gXCIgKyBpIH0pO1xuICB9O1xufSIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==) - `Function declared in a loop contains unsafe references to variable(s) 'i', 'i'. `

Is this expected, or should we print `i` just once?

@mdjermanovic @JoshuaKGoldberg ^

Reply:
Ha, nice catch. I don't know if it's expected but my personal intuition is that it's a bug.

Reply:
For tracking: #19586 -> #19595 ✔️